libs/http-service/src/http-service.cpp

```diff
@@ -108,14 +122,15 @@ struct HttpService::Impl
             self_.request(request);
         }
 
-        // Set up the streaming response
+        // Set up the streaming response.
+        // TODO explanation: what is this streaming response?
```

Comment:
How about this?

```
// Set up the response stream of tile layers as an httplib
// content provider. This allows us to push out finished
// tile layers as soon as they are ready, so the client does
// not need to wait for the server to process the whole request
// before it receives the first tile response. The content provider
// consists of two lambdas:
// (1) The first lambda is called continuously by httplib to obtain
//     potential new response data, which we are free to write into
//     the passed httplib DataSink. As our tile requests are being worked
//     on by the service, results are written into state->buffer_.
//     We pick up data from the buffer, and then pass it into the sink.
//     Once all tile requests are "done", we signal to the sink that
//     no more data is to be expected by calling sink->done().
//     This tells httplib that it may close the response stream with a 200.
//     See also httplib::write_content_without_length(...).
// (2) This lambda is called as a clean-up routine whenever the http request
//     is wrapped up by httplib. The success flag indicates whether the wrap-up was
//     caused by us calling sink->done(), or by some other external condition.
//     Such an external condition is usually a network error or an http request
//     abort signal. As tiles-requests can be *very* long-running when many
//     tiles are requested, for example in a map-viewer scenario, request aborts
//     can occur frequently, i.e. when the user moves the viewport.
```

Reply:
Shortened it a bit, but the explanation was good!